File: 888 - Fair Candy Swap.java
import java.lang.annotation.Retention;
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int[] fairCandySwap(int[] A, int[] B) {
        int at = 0, bt = 0;
        for (int i : A) {
            at += i;
        }
        for (int i : B) {
            bt += i;
        }
        int cons = (at - bt) / 2;
        Set<Integer> s = new HashSet<>();
        for (int i : A) {
            s.add(i);
        }
        int x;
        int[] res = {0, 0};
        for (int i : B) {
            x = i + cons;
            if (s.contains(x)) {
                res = new int[] {x, i};
            }
        }
        return res;
    }
}

File: 888 - Fair Candy Swap.cpp
#include <unordered_set>
#include <vector>>
using namespace std;
class Solution {
public:
    vector<int> fairCandySwap(vector<int>& A, vector<int>& B) {
        int at = 0, bt = 0;
        for (int a : A) {
            at += a;
        }
        for (int b : B) {
            bt += b;
        }
        int cons = (at - bt) / 2;
        unordered_set<int> s(A.begin(), A.end());
        vector<int> res;
        for (int y : B) {
            int x = y + cons;
            if (s.count(x) != 0) {
                res = {x, y};
            }
        }
        return res;
    }
};